### PR TITLE
Enhance README

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

This PR aims to enhance the readability of README.md.
Please check out how it looks from [my repository](https://github.com/nabenabe0928/optuna/tree/enhance-readme).

## Description of the changes

The changes are the following:
1. Embrace codeblocks with foldable tabs,
2. correct grammatical errors,
3. Highlight some important information using alerts, and
4. Add the paper information.

### CODE REVIEW

**Review by c-bata — APPROVED** [review]
Looks great to me.

File: README.md [hunk 1]
```diff
@@ -43,11 +43,16 @@ We use the terms *study* and *trial* as follows:
...
+Please refer to the sample code below. The goal of a *study* is to find out the optimal set of
 hyperparameter values (e.g., `regressor` and `svr_c`) through multiple *trials* (e.g.,
-`n_trials=100`). Optuna is a framework designed for the automation and the acceleration of the
+`n_trials=100`). Optuna is a framework designed for automation and acceleration of
 optimization *studies*.
 
+<details>
+<summary>Sample code with scikit-learn</summary>
+
+**To reviewers. we can also open here by default by using `<details open>`.**
```
Comment (nabenabe0928) [inline README.md:54]: Please check here just in case!

Comment (c-bata) [reply README.md:54]: Either option would be fine, but I prefer to open it by default.

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Pull request merged**